make_prg/interval_partition.py

```diff
@@ -0,0 +1,237 @@
+"""
+Code responsible for converting a consensus string into a set of disjoint
+match/non_match intervals.
+"""
+from enum import Enum, auto
+from typing import List, Tuple, Optional
+
+from make_prg import MSA
+
+from make_prg.seq_utils import get_interval_seqs, is_non_match, has_empty_sequence
+
+
+class PartitioningError(Exception):
+    pass
+
+
+class IntervalType(Enum):
+    Match = auto()
+    NonMatch = auto()
+
+
+def get_type(letter: str) -> IntervalType:
```

Comment:
It might make this a little more readable if this function was added as a static factory constructor on `IntervalType`?

So you would call it something like `from_char` and then when you use it, it is self documenting - `IntervalType.from_char('*')` rather than `get_type('*')` which is a bit more confusing out of context.

Or even try overriding some magic methods to remove the need for a function completely - https://stackoverflow.com/a/55259756/5299417

Reply:
lovely! adding your static constructor suggestion- prefer that than fiddling with magic methods